src/main/java/com/uber/jenkins/phabricator/tasks/GenericBuildTask.java

```diff
@@ -0,0 +1,105 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.uber.jenkins.phabricator.tasks;
+
+import com.uber.jenkins.phabricator.CodeCoverageMetrics;
+import com.uber.jenkins.phabricator.uberalls.UberallsClient;
+import com.uber.jenkins.phabricator.utils.CommonUtils;
+import com.uber.jenkins.phabricator.utils.Logger;
+import hudson.plugins.cobertura.targets.CoverageResult;
+
+/**
+ * Generic build task.
+ */
+public class GenericBuildTask extends Task {
+
+    protected UberallsClient uberallsClient;
+    protected CodeCoverageMetrics codeCoverageMetrics;
+    protected boolean uberallsEnabled;
+    protected String commitSha;
+
+    /**
+     * GenericBuildTask constructor.
+     * @param logger The logger.
+     * @param uberallsClient The uberalls client.
+     * @param codeCoverageMetrics The coverage metrics.
+     * @param uberallsEnabled Whether uberalls is enabled.
+     * @param commitSha The commit sha.
+     */
+    public GenericBuildTask(Logger logger, UberallsClient uberallsClient,
+                            CodeCoverageMetrics codeCoverageMetrics, boolean uberallsEnabled,
+                            String commitSha) {
+        super(logger);
+        this.uberallsClient = uberallsClient;
+        this.codeCoverageMetrics = codeCoverageMetrics;
+        this.uberallsEnabled = uberallsEnabled;
+        this.commitSha = commitSha;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected String getTag() {
+        return "generic-build";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void setup() {
+        // Handle bad input.
+        if (codeCoverageMetrics == null || !codeCoverageMetrics.isValid()) {
+            info("Coverage result not found. Ignoring build.");
+            result = Result.IGNORED;
+        } else if (!uberallsEnabled || CommonUtils.isBlank(uberallsClient.getBaseURL())) {
+            info("Uberalls not configured. Skipping build.");
+            result = Result.SKIPPED;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void execute() {
+        if (result == Result.UNKNWON) {
+            if (!CommonUtils.isBlank(commitSha)) {
+                info(String.format("Sending coverage result for %s as %s", commitSha,
+                        codeCoverageMetrics.toString()));
+                result = uberallsClient.recordCoverage(commitSha, codeCoverageMetrics) ?
+                        Result.SUCCESS : Result.FAILURE;
+            } else {
+                info("No line coverage found. Ignoring build.");
+                result = Result.IGNORED;
```

Comment:
serious question for us non-java folks: i see that this is defined on the superclass, but in other languages i would expect to see `this.result = `  to make it clear it's not a local. is the way you have it the preferred java style? in what cases would i use `this.` vs the implicit `this`?